apiserver/facades/agent/uniter/goal-state_test.go

```diff
@@ -0,0 +1,336 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package uniter_test
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/facades/agent/uniter"
+	"github.com/juju/juju/apiserver/params"
+	apiservertesting "github.com/juju/juju/apiserver/testing"
+	"github.com/juju/juju/juju/testing"
+	"github.com/juju/juju/state"
+	jujufactory "github.com/juju/juju/testing/factory"
```

Comment:
For this particular suite, I said it was ok (for now) to leave in JujuConnSuite since 90% of the other tests use it and it's easier to crib the existing set up code to get stuff going. I just want to unblock the landing and we can break new ground in a followup PR.